suite-common/wallet-core/src/fiat-rates/fiatRatesMiddleware.ts

```diff
@@ -78,13 +81,29 @@ export const prepareFiatRatesMiddleware = createMiddlewareWithExtraDeps(
                 updateTxsFiatRatesThunk({
                     account,
                     txs: transactions,
+                    localCurrency: prevStateLocalCurrency,
                 }),
             );
         }
 
         if (setWalletSettingsLocalCurrency.match(action)) {
-            // for coins relying on coingecko we only fetch rates for one fiat currency
-            dispatch(updateLastWeekFiatRatesThunk());
+            // for coins (ADA) depending on Coingecko API we always download fiat rates for only one currency.
+            const { localCurrency } = action.payload;
+
+            dispatch(updateLastWeekFiatRatesThunk(localCurrency));
+
+            prevStateAccounts.forEach(account => {
+                if (account.symbol !== 'ada') return;
```

Comment:
Not sure about XRP... I don't like this arbitrary condition, would be nice to have at least some config (possibly in suite-config/fiat) defining what fiat rates are not supported by blockbook and use that config here.

Reply:
I would prefer this to have in networksConfig, this is basically related to backend type right? so If coin use blockbook it should work otherwise it won't?

Reply:
the check for fiat rates provider fixed in [fe1dbc0](https://github.com/trezor/trezor-suite/pull/8642/commits/fe1dbc0c6ed9abd9a8d14b6266332fb0e919c745)